src/bootstrap/src/core/build_steps/compile.rs

```diff
@@ -346,7 +346,7 @@ fn copy_self_contained_objects(
                 panic!("Target {:?} does not have a \"wasi-root\" key", target.triple)
             })
             .join("lib")
-            .join(target.to_string().replace("-preview1", ""));
+            .join(target.to_string().replace("-preview1", "").replace("-preview2", ""));
```

Comment:
This'll want to in theroy be coordinated with @dicej's work on getting preview2 upstream with libc since the precise location of things may change

Reply:
Yeah, we actually don't want to remove the -preview2 suffix here, since the directory name will be wasm32-wasi-preview2.

Reply:
Sounds good. This is a bit of a chicken/egg problem. The hack here gets things working with the current wasi-sdk. We'll have to figure out when the changes land in the wasi-sdk and this can be ported over.